### PR TITLE
Add debugging help to docs

### PR DESCRIPTION
**Pull request by tiltingpenguin**
## Linked Items

Fixes https://github.com/cobbler/cobbler/issues/3672

 

## Description

This PR updates some of the documentation for Cobbler. It adds some context about the content to the landing page, deduplicates a section about Triggers and moves some of the autoinstallation documentation into the right section. Also it adds a note about how to debug unattended installations.

## Category

This is related to a:

- [ ] Bugfix
- [ ] Feature
- [ ] Packaging
- [x] Docs
- [ ] Code Quality
- [ ] Refactoring
- [ ] Miscellaneous

## Tests

- [ ] Unit-Tests were created
- [ ] System-Tests were created
- [ ] Code is already covered by Unit-Tests
- [ ] Code is already covered by System-Tests
- [x] No tests required

### CODE REVIEW

**Review by SchoolGuy — COMMENTED** [review]
Thanks for that cleanup.

I would mention the following topics:

- BMC with IPMI SOL
- BMC with HTML5 KVM
- A dedicated serial console
- A networked KVM (maybe mention pikvm explicitly as a diy example)

Furthermore I would give a general overview of the networked boot process with errors that can happen:

1. Firmware/EFI boot -> Wrong boot device
2. Network DHCP request from network card -> Wrong VLAN/Network, Cable issues, Firewall
3. TFTP request to the next server (typos in the Cobbler settings, inhertiance issues, daemon started but not enabled and VM was rebooted, tftp timeouts ...)
4. Kernel and Initrd start -> Missing drivers for hardware
5. HTTP requests towards Cobbler for install system and the autoinstall control file -> Firewall or proxy issues, Cobbler timeouts, Cheetah templating errors
6. Installation -> Syntax errors due to incorrect escaping, unavailable remote resources, ...
7. Reboot Loop due to netboot still being  enabled

File: docs/user-guide/automatic-installation.rst [hunk 1]
```diff
@@ -92,3 +92,49 @@ To check for potential errors in auto-installation files, prior to installation,
...
+  * Typo in cobbler settings
+  * inheritance issues
+  * VM restarted with daemon started but not enabled
+  * tftp timeout
+* Kernel & Initrd:
+  * Missing hardware drivers
+* HTTP requests towards Cobbler:
+  * Firewall/Proxy issues
+  * Cobbler timeout
+  * Cheeta tempating errors
```
Comment (SchoolGuy) [inline docs/user-guide/automatic-installation.rst:134]: Typo

File: docs/user-guide/automatic-installation.rst [hunk 2]
```diff
@@ -92,3 +92,49 @@ To check for potential errors in auto-installation files, prior to installation,
...
+may have gone interactive (or stalled/crashed) during automatic installation.
+
+Debugging of unattended installations
+#####################################
+
+There are different tools for debugging automatic installations. In general it is recommended to use something to record the 
+output of the screen, since some important information may only be visible for a short amount of time. Examples are 
+BMC (with IPMI SOL or HTML5 KVM), a dedicated serial console or a networked KVM.
+
+Here is a short list of important stages during automatic installations and errors that can occur there:
```
Comment (SchoolGuy) [inline docs/user-guide/automatic-installation.rst:116]: I would like that we clarify that this list is not complete.

**Review by SchoolGuy — COMMENTED** [review]
Much better (even though I would love if we go a bit more into detail). I found two things I think that are worth fixing before we merge the PR. :)

**Comment by tiltingpenguin** [comment]
As I don't have much experience debugging cobbler autoinstalls myself it is a bit difficult to go into details that I didn't learn from you

File: docs/user-guide/automatic-installation.rst [hunk 3]
```diff
@@ -92,3 +92,49 @@ To check for potential errors in auto-installation files, prior to installation,
...
+  * Typo in cobbler settings
+  * inheritance issues
+  * VM restarted with daemon started but not enabled
+  * tftp timeout
+* Kernel & Initrd:
+  * Missing hardware drivers
+* HTTP requests towards Cobbler:
+  * Firewall/Proxy issues
+  * Cobbler timeout
+  * Cheetah tempating errors
```
Comment (SchoolGuy) [inline docs/user-guide/automatic-installation.rst:134]:
```suggestion
  * Cheetah templating errors
```

**Review by SchoolGuy — APPROVED** [review]

**Pull request merged**